plugin/trino-apache-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1241 @@
+    {
+        for (RangerTrinoResource res : createResource(table, columns)) {
+            if (!hasPermission(res, context, SELECT, "CreateViewWithSelectFromColumns")) {
+                denyCreateViewWithSelect(table.getSchemaTableName().getTableName(), context.getIdentity());
+            }
+        }
+    }
+
+    @Override
+    public void checkCanSetViewComment(SystemSecurityContext context, CatalogSchemaTableName view)
+    {
+        if (!hasPermission(createResource(view), context, ALTER, "SetViewComment")) {
+            denyCommentView(view.toString());
+        }
+    }
+
+    @Override
+    public void checkCanCreateMaterializedView(SystemSecurityContext context, CatalogSchemaTableName materializedView, Map<String, Object> properties)
+    {
+        if (!hasPermission(createResource(materializedView), context, CREATE, "CreateMaterializedView")) {
+            denyCreateMaterializedView(materializedView.getSchemaTableName().getTableName());
+        }
+    }
+
+    @Override
+    public void checkCanRefreshMaterializedView(SystemSecurityContext context, CatalogSchemaTableName materializedView)
+    {
+        if (!hasPermission(createResource(materializedView), context, ALTER, "RefreshMaterializedView")) {
+            denyRefreshMaterializedView(materializedView.toString());
+        }
+    }
+
+    @Override
+    public void checkCanSetMaterializedViewProperties(SystemSecurityContext context, CatalogSchemaTableName materializedView, Map<String, Optional<Object>> properties)
+    {
+        if (!hasPermission(createResource(materializedView), context, ALTER, "SetMaterializedViewProperties")) {
+            denySetMaterializedViewProperties(materializedView.toString());
+        }
+    }
+
+    @Override
+    public void checkCanDropMaterializedView(SystemSecurityContext context, CatalogSchemaTableName materializedView)
+    {
+        if (!hasPermission(createResource(materializedView), context, DROP, "DropMaterializedView")) {
+            denyDropMaterializedView(materializedView.getSchemaTableName().getTableName());
+        }
+    }
+
+    @Override
+    public void checkCanRenameMaterializedView(SystemSecurityContext context, CatalogSchemaTableName materializedView, CatalogSchemaTableName newView)
+    {
+        if (!hasPermission(createResource(materializedView), context, ALTER, "RenameMaterializedView:source") || !hasPermission(createResource(newView), context, ALTER, "RenameMaterializedView:target")) {
+            denyRenameMaterializedView(materializedView.toString(), newView.toString());
+        }
+    }
+
+    @Override
+    public void checkCanGrantSchemaPrivilege(SystemSecurityContext context, Privilege privilege, CatalogSchemaName schema, TrinoPrincipal grantee, boolean grantOption)
+    {
+        if (!hasPermission(createResource(schema), context, GRANT, "GrantSchemaPrivilege")) {
```

Comment:
@mneethiraj Let's keep these methods empty. Let's not use Ranger to control access to GRANT/REVOKE statements here. We would need to implement `io.trino.plugin.hive.security.AccessControlMetadata` for Ranger in order to make use of these methods. However it is not yet part of the SPI.

Reply:
Yes, you're right @kokosing, these should be empty.

(Sorry for late response, I was on vacation)

Reply:
These methods can have meaningful implementations, but they have to be accompanied by implementation of the `SystemSecurityMetadata` interface. That's why we said that these methods should be empty.

That interface is required if you want to support things like `GRANT ... ON ... TO ...` SQL commands. But if we want to support it, it would be a separate effort. There are also commands for role management, like `CREATE ROLE ...` and `DROP ROLE ...`, but AFAICT Ranger does its own role management, so this does not make sense with this plugin.

Reply:
@ksobolew  - ok. I guess you are suggesting that a `SystemAccessControl` implementation should have empty implementation for above listed methods until it provides an implementation of `SystemSecurityMetadata` interface. Is this corect?
  
BTW, Ranger does have its own role management.

Reply:
@ksobolew, @kokosing  - can you confirm the list of methods (from the list above) whose implementation should be emptied in `RangerSystemAccessControl`?

Reply:
The list seems to be valid.